lib/pure/net.nim

```diff
@@ -567,16 +578,48 @@ when defineSsl:
     ## or using ECDSA:
     ## - ``openssl ecparam -out mykey.pem -name secp256k1 -genkey``
     ## - ``openssl req -new -key mykey.pem -x509 -nodes -days 365 -out mycert.pem``
-    var newCTX: SslCtx
-    case protVersion
-    of protSSLv23:
-      newCTX = SSL_CTX_new(SSLv23_method()) # SSlv2,3 and TLS1 support.
-    of protSSLv2:
-      raiseSSLError("SSLv2 is no longer secure and has been deprecated, use protSSLv23")
-    of protSSLv3:
-      raiseSSLError("SSLv3 is no longer secure and has been deprecated, use protSSLv23")
-    of protTLSv1:
-      newCTX = SSL_CTX_new(TLSv1_method())
+    if minProtVersion in {protSSLv2, protSSLv3}:
+      raiseSSLError("SSLv2/3 is no longer secure and has been deprecated, use protTLS")
+
+    var newCTX = SSL_CTX_new(SSLv23_method())
+    let minVersion: cint =
+      case minProtVersion
+      of protTLSv1, protSSLv23:
```

Comment:
protSSLv23 is silently using TLS1 only?

Reply:
Yes, for compatibility with the original, TLS1 is set as the lowest supported version. See https://github.com/nim-lang/Nim/pull/14556/files#diff-123d798b39756a95de82d119d2fe6661157ebb20407e2c394b7490e7588d2294R97 for the documentation.